File: tests/go/example1.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
)

func DataFlowTest(filePath string, test string) string {
	filePath := "example backward"
	newPath := filePath
	newPath := filePath
	var result string
	newPath = functionTest()

	// Vérifie si le fichier existe
	if _, err := os.Stat(newPath); os.IsNotExist(err) {
		result = "File does not exist"
	} else {
		// Lis le contenu du fichier
		content, err := ioutil.ReadFile(newPath)
		if err != nil {
			result = "Error reading file"
		} else {
			result = string(content)
		}
	}

	newPath = "test"

	return result
}

func test() {
	// Changez ceci avec le chemin du fichier que vous souhaitez tester
	filePathModified := "example.txt"
	if filePath == "" {
		fmt.Println("File does not exist")
	}
	filePath := "test"
	TEST2(filePathModified)

	filePathModified := filePathModified + 1
	test := "example.txt"
	message := DataFlowTest(filePathModified, test)

	fmt.Println(message)
}

func functionTest(filePath string) string {
	return filePath
}

func TEST2(test string) string {
	test := "example testAAA"
	return test
}

func main() {
	filePathModified := "example backward"
	test()

}

File: tests/js/example1.js
const fs = require('fs');

// Fonction pour tester le flux de données
function DataFlowTest(filePath, test) {
    filePath = "example backward";
    let newPath = filePath;
    let result = '';
    newPath = functionTest();

    // Vérifie si le fichier existe
    if (!fs.existsSync(newPath)) {
        result = "File does not exist";
    } else {
        try {
            // Lis le contenu du fichier
            const content = fs.readFileSync(newPath, 'utf8');
            result = content;
        } catch (err) {
            result = "Error reading file";
        }
    }

    newPath = "test";

    return result;
}

// Fonction d'exemple
function functionTest(filePath) {
    return filePath;
}

// Fonction de test
function test() {
    // Changez ceci avec le chemin du fichier que vous souhaitez tester
    let filePath = "example.txt";
    if (filePath === "") {
        console.log("File does not exist");
    }

    let test = "test";
    TEST2(filePathModified);

    filePathModified = filePathModified + 1;    
    // Ceci va produire une erreur car filePath est une chaîne de caractères et "filePath + 1" n'est pas valide
    test = "test"
    let message = DataFlowTest(filePathModified, test);

    console.log(message);
}

// Deuxième définition de la fonction (la redéclaration est conservée)
function functionTest(filePath) {
    return filePath;
}

// Fonction TEST2
function TEST2(test) {
    test = "example testAAA";
    return test;
}

// Fonction principale
function main() {
    let filePath = "example backward";
    test();
}

// Appel de la fonction principale
main();
